src/main/java/games/strategy/triplea/ui/ProductionPanel.java

```diff
@@ -82,20 +82,11 @@ protected ProductionPanel(final IUIContext uiContext) {
    */
   public IntegerMap<ProductionRule> show(final PlayerID id, final JFrame parent, final GameData data, final boolean bid,
       final IntegerMap<ProductionRule> initialPurchase) {
-    if (parent != null) {
```

Comment:
No, the check is being removed because it adds no value.  Leaving it in place still causes an NPE on the `dialog.pack()` line below because `dialog` will be `null`.  Since we were going to have an NPE either way, I removed it to trigger the NPE as early as possible (in `JDialogBuilder#parentFrame()`).

However, in light of your suggestion in #2184, I expect it to be added back in a forthcoming change, albeit in a slightly modified form.